src/analyze.cpp

```diff
@@ -1670,6 +1670,9 @@ static void resolve_struct_type(CodeGen *g, TypeTableEntry *struct_type) {
     if (struct_type->data.structure.is_invalid)
         return;
 
+    if (struct_type->data.structure.zero_bits_loop_flag)
```

Comment:
hmm this doesn't seem right to me. if we return from `resolve_struct_zero_bits` then either `zero_bits_loop_flag` should be `false` or `is_invalid` should be `true`. Let me poke at it.